Complete S6 MovieQuiz

Evo-Eclipse opened the pull request:
Added New Models and Services

- Added `MostPopularMovies` and `MostPopularMovie` models to represent movie data.
- Introduced `MoviesLoader` service to fetch movie data from a network source.
- Created `NetworkClient` to handle network requests.

Updates to Existing Models

- Modified `QuizQuestion` model to store image data as `Data` instead of local URL.

UI Updates

- Updated the question text in the storyboard.
- Added an activity indicator to the main storyboard for loading states

ViewController Updates

- Updated `MovieQuizViewController` to handle data loading, display loading indicators, and show network errors.

Protocol Updates

- Extended `QuestionFactoryDelegate` to include methods for handling data load success and failure.
- Updated `QuestionFactoryProtocol` to include a method for loading data.

Evo-Eclipse committed:
Complete S6 MovieQuiz

NaR0RoW commented on MovieQuiz/Models/MostPopularMovies.swift:
```diff
@@ -0,0 +1,31 @@
+    let imageURL: URL
+    
+    var resizedImageURL: URL {
+        // создаем строку из адреса
```
Стоит удалять комментарии перед review

NaR0RoW commented on MovieQuiz/Models/MostPopularMovies.swift:
```diff
@@ -0,0 +1,31 @@
+        return newURL
+    }
+    
+    private enum CodingKeys: String, CodingKey {
+    case title = "fullTitle"
+    case rating = "imDbRating"
+    case imageURL = "image"
+    }
```
Код немного поехал, можно выделить это место и воспользоваться автоматическим форматированием в Xcode `control + I`

NaR0RoW commented on MovieQuiz/Models/QuizQuestion.swift:
```diff
@@ -1,5 +1,7 @@
+
 struct QuizQuestion {
-    let image: String
+    let image: Data
```
Когда видишь слово `image`, то ожидаешь увидеть константу типа `UIImage`. Давай переименуем эту константу на `imageData`

NaR0RoW commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -41,20 +44,63 @@ final class QuestionFactory: QuestionFactoryProtocol {
             text: "Рейтинг этого фильма больше чем 6?",
             correctAnswer: false),
     ]
+    */
```
Может тогда удалим, раз этот код больше не нужен?

Evo-Eclipse replied:
> Может тогда удалим, раз этот код больше не нужен?

По ТЗ было требование оставить

Evo-Eclipse committed:
Review Fixes

- Reformatted files for consistency
- Renamed `image` to `imageData` in struct QuizQuestion for better clarity
